ac.soton.xeventb.xcontext/src/ac/soton/xeventb/xcontext/validation/XContextValidator.xtend

```diff
@@ -232,24 +234,34 @@ class XContextValidator extends AbstractXContextValidator {
 	 */
 	def private createIssue(EObject obj, IMarker rodinMarker) {
 		val attributeType = RodinMarkerUtil.getAttributeType(rodinMarker);
-		val feature = getFeature(obj, attributeType)
 		val attributes = rodinMarker.attributes
 		val severity = attributes.get(IMarker.SEVERITY)
-		// XText markers does not have information about "start" and "end"
-		// at the moment. We include this information in the message and
-		// "decode" it later when create the markers. 
-		val start = RodinMarkerUtil.getCharStart(rodinMarker)
-		val end = RodinMarkerUtil.getCharEnd(rodinMarker)
-		var message = attributes.get(IMarker.MESSAGE)
-		if (start != -1) {
-			message = message + " (from " + start + " to " + end + ")"
+        var message = attributes.get(IMarker.MESSAGE)
+		
+        var feature = getFeature(obj, attributeType)
+        var elem = obj
+        if (feature === null) {
+            elem = getContext(obj)
```

Comment:
could you use
((EventBObject)obj).getContaining(ContextPackage.Literals.CONTEXT);

Reply:
Thanks @cfsnook. I reimplemented the utility methods to use EventBObject.getContaining() method.